entrypoint/20-envsubst-on-templates.sh

```diff
@@ -10,7 +10,7 @@ auto_envsubst() {
   local output_dir="${NGINX_ENVSUBST_OUTPUT_DIR:-/etc/nginx/conf.d}"
 
   local template defined_envs relative_path output_path subdir
-  defined_envs=$(printf '${%s} ' $(env -0 | cut -z -d= -f1 | tr -d "\0"))
+  defined_envs=$(cat /proc/self/environ | xargs -I{}  -0 -n1 sh -c 'echo "{}" | sed -n "1 s/^\([^=]*\)=.*$/\${\1} /p"');
```

Comment:
With the limitations of BusyBox `grep`, I did a little "code golf" and this is the best I could come up with that seems to work on both Debian and Alpine:

```suggestion
  defined_envs=$(printf '${%s} ' $(xargs -0n1 -a /proc/self/environ sh -c 'echo "$@" | grep -oEm1 "^[^=]+"' --));
```

(Although still obviously depends on whether the maintainers want this improvement -- my goal is helping it be the best it can be if they do want it. :smile:)